tests/native-lib/fail/trace_read.stderr

```diff
@@ -0,0 +1,40 @@
+warning: integer-to-pointer cast
+  --> tests/native-lib/fail/trace_read.rs:LL:CC
+   |
+LL |     let invalid: *const i32 = std::ptr::with_exposed_provenance(intermediate.addr());
+   |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ integer-to-pointer cast
+   |
+   = help: this program is using integer-to-pointer casts or (equivalently) `ptr::with_exposed_provenance`, which means that Miri might miss pointer bugs in this program
+   = help: see https://doc.rust-lang.org/nightly/std/ptr/fn.with_exposed_provenance.html for more details on that operation
+   = help: to ensure that Miri does not miss bugs in your program, use Strict Provenance APIs (https://doc.rust-lang.org/nightly/std/ptr/index.html#strict-provenance, https://crates.io/crates/sptr) instead
+   = help: you can then set `MIRIFLAGS=-Zmiri-strict-provenance` to ensure you are not relying on `with_exposed_provenance` semantics
+   = help: alternatively, `MIRIFLAGS=-Zmiri-permissive-provenance` disables this warning
+   = note: BACKTRACE:
+   = note: inside `unexposed_reachable_alloc` at tests/native-lib/fail/trace_read.rs:LL:CC
+note: inside `main`
+  --> tests/native-lib/fail/trace_read.rs:LL:CC
+   |
+LL |     unexposed_reachable_alloc();
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: Undefined Behavior: pointer not dereferenceable: pointer must be dereferenceable for 4 bytes, but got $HEX[noalloc] which is a dangling pointer (it has no provenance)
```

Comment:
This error does not look like it has anything to do with tracing?

Reply:
It's because `let invalid: *const i32 = std::ptr::with_exposed_provenance(intermediate.addr())` creates an invalid pointer (as the alloc is not exposed). If I remove `-Zmiri-native-lib-enable-tracing`, the test xpasses with no errors